pkg/simple_cache/lib/src/providers/inmemory.dart

```diff
@@ -0,0 +1,70 @@
+import 'dart:async';
+import '../../cache_provider.dart';
+
+class _InMemoryEntry<T> {
+  final T value;
+  final DateTime _expires;
+  _InMemoryEntry(this.value, [this._expires]);
+  bool get isExpired => _expires != null && _expires.isBefore(DateTime.now());
+}
+
+/// Simple two-generational LRU cache inspired by:
+/// https://github.com/sindresorhus/quick-lru
+class InMemoryCacheProvider<T> extends CacheProvider<T> {
+  Map<String, _InMemoryEntry<T>> _new = {}; // new generation
+  Map<String, _InMemoryEntry<T>> _old = {}; // old generation
+  final int _maxSize; // maximum size before clearing old generation
+  InMemoryCacheProvider(this._maxSize);
+
+  @override
+  Future<T> get(String key) async {
+    // Lookup in the new generation
+    var entry = _new[key];
+    if (entry != null) {
+      if (!entry.isExpired) {
+        return entry.value;
+      }
+      // Remove, if expired
+      _new.remove(key);
+    }
+    // Lookup in the old generation
+    entry = _old[key];
+    if (entry != null) {
+      if (!entry.isExpired) {
+        // If not expired, we insert the entry into the new generation
+        _new[key] = entry;
+        // clear old generation, if _maxSize have been reached
+        if (_new.length >= _maxSize) {
+          _old = _new;
+          _new = {};
+        }
+        return entry.value;
+      }
+      // Remove, if expired
+      _old.remove(key);
+    }
+    return null;
+  }
+
+  @override
+  Future set(String key, T value, [Duration ttl]) async {
+    if (ttl == null) {
```

Comment:
Why not add a default ttl for the in-memory cache too? Could simplify this code and the above getter too.

Reply:
default is no `ttl`, so no expiration happens until the system runs out of memory... or whatever eviction policy the cache provider has..